### PR TITLE
Volume datasource

### PR DESCRIPTION
**Pull request by Sakshi-dell**
# Description
Volume data source support added to terraform for powerstore

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [xI have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] unit test passing successfully

### CODE REVIEW

File: models/volumeDataSource.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+type VolumeGroup struct {
+	ID          types.String `tfsdk:"id"`
+	Name        types.String `tfsdk:"name"`
+	Description types.String `tfsdk:"description"`
+}
+
+// Datastores - list of datastores for a volume in the volume data source
+type Datastores struct {
+	ID           types.String `tfsdk:"id"`
+	InstanceUUID types.String `tfsdk:"istance_uuid"`
```
Comment (shenda1) [inline models/volumeDataSource.go]: Please verify the tag for InstanceUUID

File: powerstore/datasource_volume.go [hunk 2]
```diff
@@ -0,0 +1,505 @@
...
+				Computed:            true,
+				Validators: []validator.String{
+					stringvalidator.ConflictsWith(path.MatchRoot("name")),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description:         "Name of the volume.",
+				MarkdownDescription: "Name of the volume.",
+				Optional:            true,
+				Validators: []validator.String{
```
Comment (shenda1) [inline powerstore/datasource_volume.go]: I think this validation is not required. Validation on id will take care of this. Please verify once.

File: powerstore/datasource_volume.go [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+		resp.Diagnostics.AddError(
+			"Unable to Read PowerStore Volumes",
+			err.Error(),
+		)
+		return
+	}
+	volumes = append(volumes, volume)
+	if len(volumes) == 0 {
+		resp.Diagnostics.AddError(
+			"No Volume found corresponding to the given name/ID",
```
Comment (shenda1) [inline powerstore/datasource_volume.go]: gopowerstore is not throwing an error if volume did not exist with given id/name?

File: powerstore/datasource_volume_test.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
+package powerstore
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"os"
+	"testing"
+)
+
+// Test to Create Volume
```
Comment (prajwalpatil25) [inline powerstore/datasource_volume_test.go]: comment needs to be changed

**Review by shenda1 — APPROVED** [review]

**Review by prajwalpatil25 — APPROVED** [review]

**Review by prajwalpatil25 — COMMENTED** [review]
@Sakshi-dell could you please add coverage report and I think 'dev' branch needs to be merged with this branch before merging.

**Comment by Sakshi-dell** [comment]
image

**Pull request merged**